api/providers/SupplierProvider.cs

```diff
@@ -0,0 +1,50 @@
+using DTO.Supplier;
+using FluentValidation;
+using Microsoft.EntityFrameworkCore;
+
+public class SupplierProvider : BaseProvider<Supplier>
+{
+    private ContactProvider _contactProvider;
+    private AddressProvider _addressProvider;
+    private IValidator<Supplier> _supplierValidator;
+
+    public SupplierProvider(AppDbContext db, IValidator<Supplier> validator, ContactProvider contactProvider, AddressProvider addressProvider) : base(db)
+    {
+        _supplierValidator = validator;
+        _addressProvider = addressProvider;
+        _contactProvider = contactProvider;
+    }
+
+    public override Supplier? Create(BaseDTO createValues)
+    {
+        SupplierRequest? req = createValues as SupplierRequest;
+        var allSuppliers = _db.Suppliers.Include(c => c.Contact).Include(a => a.Address).ToList();
+
+        if(allSuppliers.Any(s => s.Reference == req.Reference)) throw new ApiFlowException("Could not process create supplier request. Supplier with this reference code already exists.");
+        if (req == null) throw new ApiFlowException("Could not process create supplier request. Save new supplier failed.");
+
+        Contact? relatedContact = _contactProvider.GetOrCreateContact(req);
+        Address? relatedAddress = _addressProvider.GetOrCreateAddress(req);
+
+        Supplier newSupplier = new Supplier(newInstance: true)
+        {
+            Code = req.Code,
+            Name = req.Name,
+            Reference = req.Reference,
+            ContactId = relatedContact.Id,
+            AddressId = relatedAddress.Id,
```

Comment:
inplaats van direct deze waardes invullen ik zou hier van erna een if statement maken en als ze niet null zijn dan pas invullen als het dan komt in de validateModel functie dan indien 1 van deze ids leeg zijn pakt de validator dat op